### PR TITLE
8271490: [ppc] [s390]: Crash in JavaThread::pd_get_top_frame_for_profiling

### PR DESCRIPTION
**Pull request by RealLucy**
Please review this pull request. It hardens the code on ppc and s390 against crashes while collecting stack traces for JFR. Stack frame collection in prims/forte.cpp will benefit from this fix as well as it uses the same platform-specific code.

Testing:
SAP-internal functional and stress testing. Verifying a positive effect is hard because the targeted crashes occur only very rarely. No negative side effects of the fix have been observed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8271490](https://bugs.openjdk.java.net/browse/JDK-8271490): [ppc] [s390]: Crash in JavaThread::pd_get_top_frame_for_profiling

### Reviewers
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)
 * [Matthias Baesken](https://openjdk.java.net/census#mbaesken) (@MBaesken - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5071/head:pull/5071` \
`$ git checkout pull/5071`

Update a local copy of the PR: \
`$ git checkout pull/5071` \
`$ git pull https://git.openjdk.java.net/jdk pull/5071/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5071`

View PR using the GUI difftool: \
`$ git pr show -t 5071`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5071.diff">https://git.openjdk.java.net/jdk/pull/5071.diff</a>

</details>

### CODE REVIEW

**Comment by RealLucy** [comment]
The failing Pre-submit tests on Windows x64 are unrelated to this fix. Changes for this fix are in platform-specific files only (os_cpu/linux-ppc, os_cpu/linux-s390, cpu/ppc, cpu/s390). They can't impact Windows x64.

File: src/hotspot/os_cpu/linux_s390/thread_linux_s390.cpp
Comment (tstuefe) [inline src/hotspot/os_cpu/linux_s390/thread_linux_s390.cpp:97]: I think a reference to the JBS issue would suffice as a comment here.

**Review by tstuefe — APPROVED** [review]
LGTM. Minor comment nits.

File: src/hotspot/os_cpu/linux_s390/thread_linux_s390.cpp
Comment (RealLucy) [reply src/hotspot/os_cpu/linux_s390/thread_linux_s390.cpp:97]: Years ago, I was told that direct references to JBS bugs in the code are strongly discouraged. On the other hand, I regard it as very important to have a detailed description available. So what to do? I need a second review anyway. Maybe she has a third opinion on that.

Comment (RealLucy) [reply src/hotspot/os_cpu/linux_s390/thread_linux_s390.cpp:97]: By the way: Thanks a lot for the review!!!

**Review by MBaesken — APPROVED** [review]

**Comment by RealLucy** [comment]
Thank you, Matthias, for reviewing and approving this fix. Will integrate now.

**Pull request closed without merging**
